### PR TITLE
ReferenceStore creation, non-async

### LINKED ISSUE #202: Non-async store creation for ReferenceStore
In order to use ReferenceStore, store creation becomes `async`. This can have a big effect on the surrounding code, as the consumer and upstream call chain therefore must become async.

https://github.com/manzt/zarrita.js/blob/ee79e0a996490fd16d1efbfa395811ac0d2ab59a/packages/storage/src/ref.ts#L102

Could store creation for `ReferenceStore` be more lazy, analogous to the `ZipStore`?

https://github.com/manzt/zarrita.js/blob/ee79e0a996490fd16d1efbfa395811ac0d2ab59a/packages/storage/src/zip.ts#L82

ZipStore does not await (`this.info`) until it is needed during `get` https://github.com/manzt/zarrita.js/blob/ee79e0a996490fd16d1efbfa395811ac0d2ab59a/packages/storage/src/zip.ts#L73

### PR DESCRIPTION
**Pull request by keller-mark**
Fixes #202

### CODE REVIEW

File: packages/storage/src/ref.ts [hunk 1]
```diff
@@ -40,12 +40,12 @@ interface ReferenceStoreOptions {
...
 /** @experimental */
 class ReferenceStore implements AsyncReadable<RequestInit> {
-	#refs: Map<string, ReferenceEntry>;
+	#refs: Promise<Map<string, ReferenceEntry>>;
 	#opts: ReferenceStoreOptions;
 	#overrides: RequestInit;
 
 	constructor(
-		refs: Map<string, ReferenceEntry>,
+		refs: Promise<Map<string, ReferenceEntry>>,
```
Comment (manzt) [inline packages/storage/src/ref.ts:48]: Maybe we could make this `Promise<Map<string, ReferenceEntry>> | Map<string, ReferencEntry>`?

That way we support the previous constructor as well.

**Review by manzt — CHANGES_REQUESTED** [review]
Love this! Yeah the function coloring isn't the nicest and this is a great solution. I _think_ we could support refs as a Promise or the original, to keep the constructor signature similar to before. 

If one were to load the references separately, it might be a bit awkward to require:


```javascript
new ReferenceStore(Promise.resolve(map))
```

File: packages/storage/src/ref.ts [hunk 2]
```diff
@@ -91,19 +91,19 @@ class ReferenceStore implements AsyncReadable<RequestInit> {
 	}
 
 	static fromSpec(
-		spec: Record<string, unknown>,
+		spec: Promise<Record<string, unknown>>,
```
Comment (manzt) [inline packages/storage/src/ref.ts:94]: I think similarly we could make this

```suggestion
		spec: Promise<Record<string, unknown>> | Record<string, unknown>,
```

it's fine to `await` a non Promise in the context of `get()` which is always async.

Comment (manzt) [reply packages/storage/src/ref.ts:94]: This function could `Promise.resolve()` for consistency internally.

**Comment by keller-mark** [comment]
Just made those changes. Potentially a use case for https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/try once available as well

**Review by manzt — APPROVED** [review]
Awesome, thanks!

**Pull request merged**